File: src/main/java/fr/yodamad/svn2git/service/CleanedFilesManager.java
package fr.yodamad.svn2git.service;

import fr.yodamad.svn2git.domain.enumeration.SvnLayout;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Map;
import java.util.concurrent.atomic.AtomicLong;

import static java.lang.String.format;

public class CleanedFilesManager {

    private static final Logger LOG = LoggerFactory.getLogger(CleanedFilesManager.class);

    private final Map<String, CleanedFiles> cleanedReportMap;

    public CleanedFilesManager(Map<String, CleanedFiles> cleanedReportMap) {
        this.cleanedReportMap = cleanedReportMap;
    }

    /**
     *
     * @param beforeClean
     * @param svnLayout
     * @return true if there is at least one file in the trunk|branches|tags before|after the bfg cleaning process (calculated)
     */
    public boolean existsFileInSvnLayout(boolean beforeClean, SvnLayout svnLayout) {

        boolean existsFileAfterClean = cleanedReportMap.
            entrySet().
            stream().
            filter(entry -> svnLayout.equals(SvnLayout.ALL) ? true : entry.getValue().svnLayout.equals(svnLayout)).
            anyMatch(entry -> Boolean.TRUE.equals((beforeClean ? entry.getValue().isAtLeastOneFileBeforeClean() : entry.getValue().isAtLeastOneFileAfterClean())));

        return  existsFileAfterClean;
    }


    @Override
    public String toString() {

        StringBuilder sb = new StringBuilder();

        cleanedReportMap.
            entrySet().
            stream().
            forEach(entry -> sb.append(format("svnLocation:%s, totalFiles:%s, removedFiles:%s, remainingFiles:%s \n",
                entry.getKey(), entry.getValue().fileCountBeforeClean, entry.getValue().deletedFileCountAfterClean, entry.getValue().fileCountAfterClean)));

        return sb.toString();
    }

    public long getFileSizeTotalBeforeClean() {

        AtomicLong total = new AtomicLong(0L);

        cleanedReportMap.
            entrySet().
            stream().
            forEach(entry -> total.addAndGet(entry.getValue().fileSizeTotalBeforeClean));

        return total.get();
    }

    public long getFileSizeTotalAfterClean() {

        AtomicLong total = new AtomicLong(0L);

        cleanedReportMap.
            entrySet().
            stream().
            forEach(entry -> total.addAndGet(entry.getValue().fileSizeTotalAfterClean));

        return total.get();

    }

    /**
     * Gets cleanedReportMap
     *
     * @return value of cleanedReportMap
     */
    public Map<String, CleanedFiles> getCleanedReportMap() {
        return cleanedReportMap;
    }
}

File: src/main/kotlin/fr/yodamad/svn2git/service/CleanedFilesManager.kt
package fr.yodamad.svn2git.service

import fr.yodamad.svn2git.data.CleanedFiles
import fr.yodamad.svn2git.domain.enumeration.SvnLayout
import java.util.concurrent.atomic.AtomicLong

open class CleanedFilesManager(val cleanedReportMap: Map<String, CleanedFiles>) {

    /**
     *
     * @param beforeClean
     * @param svnLayout
     * @return true if there is at least one file in the trunk|branches|tags before|after the bfg cleaning process (calculated)
     */
    fun existsFileInSvnLayout(beforeClean: Boolean, svnLayout: SvnLayout): Boolean {
        return cleanedReportMap.entries.stream().filter { entry: Map.Entry<String, CleanedFiles> ->
            if (svnLayout == SvnLayout.ALL) true
            else entry.value.svnLayout == svnLayout }.anyMatch {
                entry: Map.Entry<String, CleanedFiles> ->
                    if (beforeClean) entry.value.isAtLeastOneFileBeforeClean
                    else entry.value.isAtLeastOneFileAfterClean
            }
    }

    override fun toString(): String {
        val sb = StringBuilder()
        cleanedReportMap.entries.stream().forEach { entry: Map.Entry<String, CleanedFiles> ->
            sb.append(String.format("svnLocation:%s, totalFiles:%s, removedFiles:%s, remainingFiles:%s \n",
                entry.key, entry.value.fileCountBeforeClean, entry.value.deletedFileCountAfterClean, entry.value.fileCountAfterClean))
        }
        return sb.toString()
    }

    val fileSizeTotalBeforeClean: Long
        get() {
            val total = AtomicLong(0L)
            cleanedReportMap.entries.stream().forEach { entry: Map.Entry<String, CleanedFiles> -> total.addAndGet(entry.value.fileSizeTotalBeforeClean) }
            return total.get()
        }
    val fileSizeTotalAfterClean: Long
        get() {
            val total = AtomicLong(0L)
            cleanedReportMap.entries.stream().forEach { entry: Map.Entry<String, CleanedFiles> -> total.addAndGet(entry.value.fileSizeTotalAfterClean) }
            return total.get()
        }
}
